web/gds-admin-ui/src/components/TopbarSearch.js

```diff
@@ -108,38 +96,77 @@ const MenuList = (props) => {
 
 /* fomates the option label */
 const handleFormatOptionLabel = (option) => {
-    const formattedOption = optionGetter(option);
-    return <div>{formattedOption}</div>;
+    return <Link className='text-muted d-block' to={`/vasps/${option.value}`}>{option.label}</Link>;
 };
 
-type SearchResultItem = {
-    id: number,
-    title: string,
-    redirectTo: string,
-    icon: string,
-};
 
-type TopbarSearchProps = {
-    items: Array<SearchResultItem>,
-};
+function formateOptions(options) {
+    return Object.entries(options).map(([k, v]) => ({ label: k, value: v }))
+}
 
-const TopbarSearch = (props: TopbarSearchProps): React$Element<any> => {
 
-    const onClick = (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-    };
+const TopbarSearch = (props) => {
+    const [options, setOptions] = React.useState([])
+    const [inputValue, setInputValue] = React.useState('')
+    const dispatch = useDispatch()
+    const safeDispatch = useSafeDispatch(dispatch)
+    const autocomplete = useSelector(fetchAllAutocomplete)
+    const [isLoading, setIsLoading] = React.useState(false)
+    const [menuIsOpen, setMenuIsOpen] = React.useState(false)
+
+
+    React.useEffect(() => {
+        safeDispatch(fetchAutocomplete())
+    }, [safeDispatch])
+
+    const filteredOptions = (input = '') => {
+        const f = formateOptions(autocomplete)
+        return f.filter((option) => option.value.toLowerCase().includes(input.toLowerCase()) || option.label.toLowerCase().includes(input.toLowerCase()))
+    }
+
+    const loadOptions = input => {
+        return new Promise((resolve, reject) => {
+            if (input.length < 2) {
+                return resolve([])
+            }
+
+            setTimeout(() => {
+                return resolve(filteredOptions(input))
+            }, 500)
```

Comment:
Is this throttling to prevent a huge number of API requests as the user types?

Reply:
No, it’s to make it a little asynchronous since I receive all the data in one request

Reply:
No, it’s to make it a little asynchronous since I receive all the data in one request

Reply:
Ok, I would refer to that as throttling - but I'm glad it's in the code!